File: subprojects/java-compatibility/src/test/java/kotools/types/internal/KotoolsTypesVersionTest.java
/*
 * Copyright 2023 Kotools S.A.S.U.
 * Use of this source code is governed by the MIT license.
 */

package kotools.types.internal;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.Arrays;

public class KotoolsTypesVersionTest {
    @Test
    public void toString_should_pass() {
        final KotoolsTypesVersion[] versions = KotoolsTypesVersion.values();
        Arrays.stream(versions).forEach(version -> {
            final String actual = version.toString();
            final String expected = version.name()
                    .substring(1)
                    .replace('_', '.');
            Assertions.assertEquals(expected, actual);
        });
    }
}

File: subprojects/internal/src/commonTest/kotlin/KotoolsTypesVersionTest.kt
/*
 * Copyright 2023 Kotools S.A.S.U.
 * Use of this source code is governed by the MIT license.
 */

package kotools.types.internal

import kotlin.test.Test
import kotlin.test.assertEquals

class KotoolsTypesVersionTest {
    @Test
    fun toString_should_pass(): Unit = KotoolsTypesVersion.values().forEach {
        val actual = "$it"
        val expected: String = it.name.drop(1)
            .replace('_', '.')
        assertEquals(expected, actual)
    }
}
